src/overlays/actors/ovl_En_Door/z_en_door.c

```diff
@@ -57,63 +97,155 @@ ActorInit En_Door_InitVars = {
+    /* 24 */ DOOR_OBJINFO_24,                         // ENDOOR_SCH_TYPE_TOWN_SHOOTING_GALLERY
+    /* 25 */ DOOR_OBJINFO_25,                         // ENDOOR_SCH_TYPE_TREASURE_CHEST_SHOP
+    /* 26 */ DOOR_OBJINFO_26,                         // ENDOOR_SCH_TYPE_HONEY_DARLING_SHOP
+    /* 27 */ DOOR_OBJINFO_27,                         // ENDOOR_SCH_TYPE_MILK_BAR
+    /* 28 */ DOOR_OBJINFO_28,                         // ENDOOR_SCH_TYPE_INN_MAIN_ENTRANCE
+    /* 29 */ DOOR_OBJINFO_29,                         // ENDOOR_SCH_TYPE_INN_UPPER_ENTRANCE
+    /* 30 */ DOOR_OBJINFO_30,                         // ENDOOR_SCH_TYPE_INN_GRANNYS
+    /* 31 */ DOOR_OBJINFO_31,                         // ENDOOR_SCH_TYPE_INN_STAFF_ROOM
+    /* 32 */ DOOR_OBJINFO_32,                         // ENDOOR_SCH_TYPE_INN_KNIFE_CHAMBER
+    /* 33 */ DOOR_OBJINFO_33,                         // ENDOOR_SCH_TYPE_INN_LARGE_SUITE
+    /* 34 */ DOOR_OBJINFO_34,                         // ENDOOR_SCH_TYPE_MAYORS_RESIDENCE_MAIN_ENTRANCE
+    /* 35 */ DOOR_OBJINFO_35,                         // ENDOOR_SCH_TYPE_MAYORS_RESIDENCE_MAYOR_DOTOUR
+    /* 36 */ DOOR_OBJINFO_36,                         // ENDOOR_SCH_TYPE_MAYORS_RESIDENCE_MADAME_AROMA
+    /* 37 */ DOOR_OBJINFO_37,                         // ENDOOR_SCH_TYPE_MAYORS_RESIDENCE_BEDROOM
+    /* 38 */ DOOR_OBJINFO_38,                         // ENDOOR_SCH_TYPE_21
+    /* 39 */ DOOR_OBJINFO_39,                         // ENDOOR_SCH_TYPE_ROMANI_RANCH_MAMAS_HOUSE
+    /* 40 */ DOOR_OBJINFO_40,                         // ENDOOR_SCH_TYPE_ROMANI_RANCH_BARN
+    /* 41 */ DOOR_OBJINFO_41,                         // ENDOOR_SCH_TYPE_ROMANI_RANCH_CUCCO_SHACK
+    /* 42 */ DOOR_OBJINFO_42,                         // ENDOOR_SCH_TYPE_ROMANI_RANCH_DOGGY_RACETRACK
+    /* 43 */ DOOR_OBJINFO_43,                         // ENDOOR_SCH_TYPE_ROMANI_RANCH_BEDROOM
+    /* 44 */ DOOR_OBJINFO_44,                         // ENDOOR_SCH_TYPE_IKANA_CANYON_MUSIC_BOX_HOUSE
+    /* 45 */ DOOR_OBJINFO_45,                         // ENDOOR_SCH_TYPE_DAMPES_HOUSE
+    /* 46 */ DOOR_OBJINFO_46,                         // ENDOOR_SCH_TYPE_MAGIC_HAGS_POTION_SHOP
+    /* 47 */ DOOR_OBJINFO_47,                         // ENDOOR_SCH_TYPE_30
+    /* 48 */ DOOR_OBJINFO_48,                         // ENDOOR_SCH_TYPE_SWAMP_SHOOTING_GALLERY
+    /* 49 */ DOOR_OBJINFO_MAX
+} EnDoorObjectInfoIndex;
+
+// These static asserts try to ensure the two enums don't get out of sync
+static_assert(ENDOOR_SCH_TYPE_SWORDSMANS_SCHOOL == DOOR_OBJINFO_17 - DOOR_OBJKIND_SCHEDULE,
+              "The enums values of `EnDoorScheduleType` and `EnDoorObjectInfoIndex` (from `DOOR_OBJKIND_SCHEDULE` "
+              "onwards) must be synced.");
+static_assert(ENDOOR_SCH_TYPE_MAX == DOOR_OBJINFO_MAX - DOOR_OBJKIND_SCHEDULE,
+              "The enums values of `EnDoorScheduleType` and `EnDoorObjectInfoIndex` (from `DOOR_OBJKIND_SCHEDULE` "
+              "onwards) must be synced.");
```

Comment:
Here I check for the start of the schedule range of `EnDoorObjectInfoIndex` and the max of this enum to be synced with the `EnDoorScheduleType` enum.
So if one enum gets an extra entry without the other one being updated, the compiler will bark at you